src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -367,14 +371,38 @@ private void initializeFromClientConfig(JedisClientConfig config) {
     }
   }
 
-  private String auth(final String password) {
-    sendCommand(Protocol.Command.AUTH, password);
-    return getStatusCodeReply();
+  private void auth(final RedisCredentialsProvider credentialsProvider) {
+    credentialsProvider.prepare();
+    auth((Supplier<RedisCredentials>) credentialsProvider);
+    credentialsProvider.cleanUp();
   }
 
-  private String auth(final String user, final String password) {
-    sendCommand(Protocol.Command.AUTH, user, password);
-    return getStatusCodeReply();
+  private void auth(final Supplier<RedisCredentials> credentialsProvider) {
+    RedisCredentials credentials = credentialsProvider.get();
+    if (credentials == null) return;
+    char[] password = credentials.getPassword();
+    if (password != null) {
+      String user = credentials.getUser();
+      byte[] binaryPassword = toBytes(password);
+      if (user != null) {
```

Comment:
WDYT ```if user != null && user != ""```

Looking at redis.clients.jedis.RedisCredentials, isn't it possible for an implementor to pass in an empty username?

Reply:
We don't do such checks in Jedis and leave it for the users to do their own thing.

(If I recall correctly,) Once there was such check for password which _had to be_ withdrawn due to issues.